File: LibraryApp/app/src/main/java/com/example/libraryapp/LostItem.java
package com.example.libraryapp;

public class LostItem {
    private String id;
    private String description;
    private String dateFound;
    private String locationFound;
    private String status; // "Unclaimed" or "Claim Pending"

    public LostItem() {} // Required for Firebase

    public LostItem(String id, String description, String dateFound, String locationFound, String status) {
        this.id = id;
        this.description = description;
        this.dateFound = dateFound;
        this.locationFound = locationFound;
        this.status = status;
    }

    public String getId() { return id; }
    public String getDescription() { return description; }
    public String getDateFound() { return dateFound; }
    public String getLocationFound() { return locationFound; }
    public String getStatus() { return status; }
}

File: src/main/java/com/example/libraryapp20/LostItem.kt
package com.example.libraryapp20

/**
 * Model for lost & found items.
 * Keeps a no-arg constructor so Firestore can deserialize documents.
 */
class LostItem() {

    var id: String? = null
    var description: String? = null
    var dateFound: String? = null
    var locationFound: String? = null
    var status: String? = null // "Unclaimed" or "Claim Pending"

    // Full constructor for convenience when creating instances in code
    constructor(
        id: String?,
        description: String?,
        dateFound: String?,
        locationFound: String?,
        status: String?
    ) : this() {
        this.id = id
        this.description = description
        this.dateFound = dateFound
        this.locationFound = locationFound
        this.status = status
    }
}
